rocketc/rocketc.py

```diff
@@ -589,20 +621,21 @@ def leave_group(self, data, suffix=""):
         if group_name == self.team_channel or group_name == self.user_data["course"]:
             return {"success": False, "error": "You Can Not Leave a Main Group"}
 
-        return self._remove_user_from_group(group_name, user["user"]["_id"])
+        return self._remove_user_from_group(group_name, user["user"]["_id"], api)
 
     @XBlock.json_handler
     def get_list_of_groups(self, data, suffix=""):
         """Returns a list with the group names"""
         # pylint: disable=unused-argument
         user_id = data.get("userId", None)
         auth_token = data.get("authToken", None)
+        self.api_teams = self._api_teams()  # pylint: disable=attribute-defined-outside-init
```

Comment:
just to clarify, when you're in a method decorated as json_handler do you have the api clients unset? if they're unset what's your approach? defining a local var and use it during the method?:
`api_teams = self._api_teams()` or set the xblock instance attribute with the api (as it is in this case)?
`self.api_teams = self._api_teams()`

Reply:
First question - yes , and my approach is the second because there are methods inside that needs this api, if  I don't set this I should pass the api like in remove_user_from_group